Week04/lesson08/homework/temp_converter/js/index.js

```diff
@@ -0,0 +1,15 @@
+function convert () {
+var f;
+var c = 20;
+
+document.getElementById("#fahrenheit").value = fahrenheit;
+document.getElementById("#celsius").value = celsius;
+
+  if (f = c*9/5+32) {
+    convert = "#celsius";
+  } else if (c = f-32*5/9) {
+    convert = "#fahrenheit";
+}
```

Comment:
Ok, so first things first, indentation is important in JS too, it helps us locate bugs (like the "Unexpected end of input" error I'm seeing in the console the minute I load up your HW), so let's fix it:

```
function convert () {
  var f;
  var c = 20;

  document.getElementById("#fahrenheit").value = fahrenheit;
  document.getElementById("#celsius").value = celsius;

  if (f = c*9/5+32) {
    convert = "#celsius";
  } else if (c = f-32*5/9) {
    convert = "#fahrenheit";
}
```

... now that we've fixed the indentation, it's easier to see that the issue is the closing of the if/else statement. There's no closing `}`, let's fix that:

```
if (f = c*9/5+32) {
  convert = "#celsius";
} else if (c = f-32*5/9) {
  convert = "#fahrenheit";
}
```

... and in context that looks like:

```
function convert () {
  var f;
  var c = 20;

  document.getElementById("#fahrenheit").value = fahrenheit;
  document.getElementById("#celsius").value = celsius;

  if (f = c*9/5+32) {
    convert = "#celsius";
  } else if (c = f-32*5/9) {
    convert = "#fahrenheit";
  }    //this to the left is the missing bracket
}
```

Cool. Now that we've not got any errors upon loading the page, let's take a deeper look into the code:

```
document.getElementById("#fahrenheit").value = fahrenheit;
document.getElementById("#celsius").value = celsius;
```

So, you've clearly got the idea right in the sense that you want to store these two values you pull off the DOM. BUT you've forgotten how we declare and assign variables. As per the slides, here's an example of variable declaration and assignment:

`var age = 21;`

So we need to apply this same way of declaring variables to your first two lines:

```
var fahrenheit = document.getElementById("#fahrenheit").value;
var celsius = document.getElementById("#celsius").value;
```

You may be thinking "duh! That's what I meant!" and I get your frustration. Totally. But, though it's clear to you that `some_stuff_to_get_the_value_out_of_the_dom = name_of_variable`, JS doesn't understand variable declaration and assignment that way. It NEEDS that `var` keyword and it NEEDS the declaring of the variable name to be on the left of the `=` sign.

Next let's take a look at your `if/else` statement. Remember, within the parentheses of `if () { //do stuff }` you need to COMPARE something, you need to establish whether something is TRUE or FALSE. What you've done here is accidentally assign `f` to `c*9/5+32`. You're not checking if something is TRUE or FALSE.

However, what to check for? Well, you need to know whether the user typed into the `c` input (and therefore is looking to convert to `f`) or if the user typed into the `f` input (and therefore is looking to convert to `c`) right? So that's something you can CHECK for. So in pseudocode:

```
if (user_filled_in_c_input) {
  //var result = conversion_of_c_into_f
} else {
  //var result = conversion_of_f_into_c
}
```

By the way, you don't need an `else if (condition_here)` because since there's only two inputs it's pretty safe to assume that if the user *didn't* fill in the `c` input, they therefore filled in the `f` input.

Right. So HOW to check if `user_filled_in_c_input`. If only we had the value of the `c` input stored somewhere... oh wait! We DO! In the new variable `celsius` we just created! So basically, the `c` input will be considered filled as long as the user typed in SOMETHING into the input. In other words, we need to make sure the `c` input (whose value is stored in the variable `celsius`) is NOT empty. We also need to somehow store the result of the conversion (like with a new variable):

```
if (celsius !== "") {
  var result = c * 9/5 + 32;
} else {
  var result = (f - 32) * 5/9;
}
```

So, all in all, you've now got:

```
function convert () {
  var f;
  var c = 20;

  var fahrenheit = document.getElementById("#fahrenheit").value;
  var celsius = document.getElementById("#celsius").value;

  if (celsius !== "") {
    var result = c * 9/5 + 32;
  } else {
    var result = (f - 32) * 5/9;
  }
}
```

... and as per link 14 in your current code below, the `convert()` function will get triggered upon click of the button.

Nearly done.

Now, you need to write back `result` to the DOM, at the very end of this function, before it closes. But I'll leave that one for you to figure out 😉